src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/ErrorFixerAgent.cs

```diff
@@ -0,0 +1,93 @@
+    {
+        private readonly AppSettings AppSettings;
+        private readonly ILogger<ErrorFixerAgent> Logger;
+        private readonly PersistentAgentsAdministrationClient AdminClient;
+        private readonly Lazy<PersistentAgent> Agent;
+
+        public ErrorFixerAgent(
+            AppSettings appSettings,
+            ILogger<ErrorFixerAgent> logger,
+            PersistentAgentsAdministrationClient adminClient)
+        {
+            AppSettings = appSettings ?? throw new ArgumentNullException(nameof(appSettings));
+            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            AdminClient = adminClient ?? throw new ArgumentNullException(nameof(adminClient));
+            Agent = new Lazy<PersistentAgent>(() => CreateAgent());
+        }
+
+        public async Task FixCodeAsync(CancellationToken ct)
+        {
+            PersistentAgent agent = Agent.Value;
+
+            // TODO: Implement the code fixing logic here
+            await Task.CompletedTask.ConfigureAwait(false);
+        }
+
+        private PersistentAgent CreateAgent()
+        {
+            Logger.LogInformation("Creating AZC Fixer agent...");
+            
+            Response<PersistentAgent> response = AdminClient.CreateAgent(
+                model: AppSettings.Model,
+                name: AppSettings.AgentName,
+                instructions: AppSettings.AgentInstructions,
+                tools: new[] { new FileSearchToolDefinition() });
+
+            PersistentAgent agent = response.Value;
+            if (agent == null || string.IsNullOrEmpty(agent.Id))
+            {
+                throw new InvalidOperationException("Failed to create AZC Fixer agent");
+            }
+
+            Logger.LogInformation("✅ Agent created successfully: {Name} ({Id})", agent.Name, agent.Id);
+            return agent;
+        }
+
+        private async Task DeleteAgentsAsync(CancellationToken ct)
+        {
+            List<Task> deleteTasks = new List<Task>();
+
+            await foreach (PersistentAgent agent in AdminClient.GetAgentsAsync(cancellationToken: ct))
+            {
+                Task deleteTask = AdminClient.DeleteAgentAsync(agent.Id, ct)
+                    .ContinueWith(t => 
+                    {
+                        if (t.IsFaulted)
+                        {
+                            Logger.LogError(t.Exception, "Failed to delete agent {Name} ({Id})", agent.Name, agent.Id);
+                        }
+                        else
+                        {
+                            Logger.LogInformation("Deleted agent: {Name} ({Id})", agent.Name, agent.Id);
+                        }
+                    }, ct);
+
+                deleteTasks.Add(deleteTask);
+            }
+
+            await Task.WhenAll(deleteTasks).ConfigureAwait(false);
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            if (Agent.IsValueCreated)
```

Comment:
As written, this is safe to call multiple times, but inefficient to do.  `Dispose` should be idempotent.  You can tweak this by just adding a `_disposed` member or swapping your `Lazy` with `null`..   

The rest of the methods would need to check for "am I disposed" and throw, if you do

Not very impactful here, because it's a closed scope and your implementation owns it -end-to-end.   I don't feel strongly one way or the other about it, so feel free to make the call that you think is best!.